-Ycheck:inlinedPositions failure with Tuples and multiple inlined calls

The simplest reproduction I could come up with,
```scala
object Test {
  inline def s1 = 47 *: s2

  inline def s2 = 48 *: s3

  inline def s3 = 49 *: ()

  s1
}
```
fails with,
```
java.lang.AssertionError: assertion failed: wrong source set for 49 # 1264 of class dotty.tools.dotc.ast.Trees$Literal, set to tests/pos/ixxxx.scala but context had /home/miles/projects/dotty/library/src-3.x/scala/Tuple.scala while compiling tests/pos/ixxxx.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: wrong source set for 49 # 1264 of class dotty.tools.dotc.ast.Trees$Literal, set to tests/pos/ixxxx.scala but context had /home/miles/projects/dotty/library/src-3.x/scala/Tuple.scala
```
The original was a (fairly simple) recursive case,
```scala
import scala.compiletime._

object Test {
  inline def summonValues[T] <: Tuple = inline erasedValue[T] match {
    case _: Unit => ()
    case _: (a *: b) => constValue[a] *: summonValues[b]
  }

  summonValues[(96, 97, 98)]
}
```
which fails similarly with,
```
java.lang.AssertionError: assertion failed: wrong source set for 98 # 1605 of class dotty.tools.dotc.ast.Trees$Literal, set to tests/pos/ixxxx.scala but context had /home/miles/projects/dotty/library/src-3.x/scala/Tuple.scala while compiling tests/pos/ixxxx.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: wrong source set for 98 # 1605 of class dotty.tools.dotc.ast.Trees$Literal, set to tests/pos/ixxxx.scala but context had /home/miles/projects/dotty/library/src-3.x/scala/Tuple.scala
```

--- a/compiler/test/dotc/pos-test-pickling.blacklist
+++ b/compiler/test/dotc/pos-test-pickling.blacklist
@@ -15,6 +15,7 @@ i8052.scala
 tuple-filter.scala
 i7740a.scala
 i7740b.scala
+i6507b.scala
 
 # Stale symbol: package object scala
 seqtype-cycle

--- a/tests/pos/i6507b.scala
+++ b/tests/pos/i6507b.scala
@@ -1,0 +1,10 @@
+import scala.compiletime._
+
+object Test {
+  transparent inline def summonValues[T]: Tuple = inline erasedValue[T] match {
+    case _: EmptyTuple => EmptyTuple
+    case _: (a *: b) => constValue[a] *: summonValues[b]
+  }
+
+  summonValues[(96, 97, 98)]
+}

